src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -139,108 +147,136 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
-                {
-                    var inputs = new StringBuilder();
-                    foreach (var parameter in message.Parameters)
-                    {
-                        var name = HtmlEncoder.Encode(parameter.Key);
-                        var value = HtmlEncoder.Encode(parameter.Value);
 
-                        var input = string.Format(CultureInfo.InvariantCulture, InputTagFormat, name, value);
-                        inputs.AppendLine(input);
-                    }
+                var issuer = HtmlEncoder.Encode(message.IssuerAddress);
 
-                    var issuer = HtmlEncoder.Encode(message.IssuerAddress);
+                var content = string.Format(CultureInfo.InvariantCulture, HtmlFormFormat, issuer, inputs);
+                var buffer = Encoding.UTF8.GetBytes(content);
 
-                    var content = string.Format(CultureInfo.InvariantCulture, HtmlFormFormat, issuer, inputs);
-                    var buffer = Encoding.UTF8.GetBytes(content);
+                Response.ContentLength = buffer.Length;
+                Response.ContentType = "text/html;charset=UTF-8";
 
-                    Response.ContentLength = buffer.Length;
-                    Response.ContentType = "text/html;charset=UTF-8";
+                // Emit Cache-Control=no-cache to prevent client caching.
+                Response.Headers[HeaderNames.CacheControl] = "no-cache";
+                Response.Headers[HeaderNames.Pragma] = "no-cache";
+                Response.Headers[HeaderNames.Expires] = "-1";
 
-                    // Emit Cache-Control=no-cache to prevent client caching.
-                    Response.Headers[HeaderNames.CacheControl] = "no-cache";
-                    Response.Headers[HeaderNames.Pragma] = "no-cache";
-                    Response.Headers[HeaderNames.Expires] = "-1";
+                await Response.Body.WriteAsync(buffer, 0, buffer.Length);
+            }
+            else
+            {
+                throw new NotImplementedException($"An unsupported authentication method has been configured: {Options.AuthenticationMethod}");
+            }
+        }
+
+        /// <summary>
+        /// Response to the callback from OpenId provider after session ended.
+        /// </summary>
+        /// <returns>A task executing the callback procedure</returns>
+        protected virtual async Task<bool> HandleSignOutCallbackAsync()
+        {
+            StringValues protectedState;
+            if (Request.Query.TryGetValue(OpenIdConnectParameterNames.State, out protectedState))
+            {
+                var properties = Options.StateDataFormat.Unprotect(protectedState);
 
-                    await Response.Body.WriteAsync(buffer, 0, buffer.Length);
+                var redirectContext = new RedirectContext(Context, Options, properties);
+                await Options.Events.RedirectToPostLogoutRedirectUri(redirectContext);
+                if (redirectContext.HandledResponse || redirectContext.Skipped)
```

Comment:
Note: you can't return `true` for the `Skipped` case (because the whole point of `SkipToNextMiddleware()` is to bypass the built-in logic and to allow the rest of the pipeline to handle the response: returning `true` will prevent that)